src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -293,201 +298,211 @@ private async Task<SearchResult> SearchImpl(SqlSearchOptions sqlSearchOptions, S
                                                .AcceptVisitor(IncludeRewriter.Instance)
                                            ?? SqlRootExpression.WithResourceTableExpressions();
 
-            using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, true))
-            using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateRetrySqlCommand())
+            using (SqlCommand sqlCommand = new SqlCommand()) // WARNING, this code will not set sqlCommand.Transaction. Sql transactions via C#/.NET are not supported in this method.
             {
                 var exportTimeTravel = clonedSearchOptions.QueryHints != null && _schemaInformation.Current >= SchemaVersionConstants.ExportTimeTravel;
                 if (exportTimeTravel)
                 {
-                    PopulateSqlCommandFromQueryHints(clonedSearchOptions, sqlCommandWrapper);
-                    sqlCommandWrapper.CommandTimeout = 1200; // set to 20 minutes, as dataset is usually large
+                    PopulateSqlCommandFromQueryHints(clonedSearchOptions, sqlCommand);
+                    sqlCommand.CommandTimeout = 1200; // set to 20 minutes, as dataset is usually large // TODO: read from same place where other timeout is read from (settings?)
```

Comment:
>// TODO: read from same place where other timeout is read from (settings?) [](http://example.com/codeflow?start=104&length=74)

There are many places where individual timeouts per command and per usage are set. Common settings does not sound like good place for this data.

Reply:
I think things like timings are usually set in configs, not hard coded into the source code.

Reply:
See me response below. This is not that trivial.